### PR TITLE
Always use UInt32 for non-inline relationship fields, but keep the client metadata for hotfixes. (Fixes #9)

### LINKED ISSUE #9: DB structure should be ignored for ref IDs
As initially reported by Shauren on IRC, it appears that there's an issue with meta sometimes being incorrect for reference IDs. For the WoW client this won't affect it until hotfixes are applied for the larger IDs, but will when they try to hotfix something bigger than what the type is able to contain. Said bug has been reported to Blizz already.

[AzeriteTierUnlock](https://github.com/wowdev/WoWDBDefs/blob/b97a4401af3ea22bab4ee6be899d297a4e08c1a4/definitions/AzeriteTierUnlock.dbd#L14) recently got a reference index value to 274, which is larger than a `uint8` can contain. DBCD crashes here:

https://github.com/wowdev/DBCD/blob/898fb2b458ec4e7ea65847dea94b2b1535fd15ac/DBFileReaderLib/Readers/WDC3Reader.cs#L104 with a rather expected `System.OverflowException: 'Value was either too large or too small for an unsigned byte.'`. 

The most obvious suggestion would be using the original reference ID type of `uint32` instead of trying to convert it to fit in with meta, but this _might_ cause issues for applying hotfixes to reference fields which might need to be handled differently.

### PR DESCRIPTION
**Pull request by justMaku**

### CODE REVIEW

File: DBCD/DBCDBuilder.cs [hunk 1]
```diff
@@ -83,7 +83,17 @@ internal DBCDBuilder(Locale locale = Locale.None)
                 var columnDefinition = databaseDefinition.columnDefinitions[fieldDefinition.name];
                 bool isLocalisedString = columnDefinition.type == "locstring" && locStringSize > 1;
 
-                var fieldType = FieldDefinitionToType(fieldDefinition, columnDefinition, localiseStrings);
+
+                Type fieldType;
+                if (fieldDefinition.isRelation && fieldDefinition.isRelation)
```
Comment (Marlamin) [inline DBCD/DBCDBuilder.cs]: Probably this?

```suggestion
                if (fieldDefinition.isRelation && fieldDefinition.isNonInline)
```

**Comment by Marlamin** [comment]
Other than above comment, looks good, fixes the issue and runs fine in production.

File: DBCD/DBCDBuilder.cs (hunk 1 above)
Comment (justMaku) [reply DBCD/DBCDBuilder.cs]: That'a actually supposed to be `isArray`, but yes, an oof.

Comment (justMaku) [reply DBCD/DBCDBuilder.cs]: Actually, scratch that, not needed at all. Oof has been oofed.

**Comment by Marlamin** [comment]
@barncastle Can you take a peek at this? Am in a bit of a conflict pickle in running this and latest master.

**Comment by Marlamin** [comment]
image
For what its worth, this is the change I'm currently running. Maku's fixes should be more broad and deal with some of the side effects of doing that.

**Comment by barncastle** [comment]
It took me far too long to understand the issue - I've been out the game too long!

Everything looks fine and appears to be working correctly. Elegantly done too I might add!

**Pull request merged**